src/transform.ts

```diff
@@ -157,14 +227,21 @@ export class Transform {
   }
 }
 
-function insertSortedAndValidate(transformations: Transformation[], t: Transformation): void {
+function insertSortedAndValidate(
+  transformations: Transformation[],
+  t: Transformation,
+  error: (byteIndex: number, msg: string) => Error,
+): void {
   transformations.push(t);
   transformations.sort(compareTransformations); // checks for overlaps
   for (let i = transformations.indexOf(t) + 1; i < transformations.length; i += 1) {
     const s = transformations[i];
     const c = compareContainment(t, s);
     if (typeof c === 'number' && c < 0) {
-      throw new Error(`A bigger area has already been transformed (${s.kind} > ${t.kind})`);
+      throw error(s.start, `A bigger area has already been transformed (${s.kind} > ${t.kind})`);
     }
   }
 }
+function byteToLineNumber(buf: Buffer, byteIndex: number): number {
+  return 1 + [...buf.slice(0, byteIndex).toString('utf8').matchAll(/\n/g)].length;
```

Comment:
Is that more efficient than
```suggestion
  return buf.slice(0, byteIndex).toString('utf8').split('\n').length;
```
?

Reply:
It's probably the same. I'll accept the suggestion.